File: frontend/src/types/draft.ts
export type DraftStatus = "generated" | "approved" | "rejected" | "sent";

export interface Draft {
  id: string;
  email_id: string;
  body: string;
  status: DraftStatus;
  created_at: string;
  updated_at: string;
}

export interface DraftCreateRequest {
  email_id: string;
  instructions?: string | null;
}

export interface DraftUpdateRequest {
  body: string;
}

File: backend/app/presentation/api/v1/schemas/draft.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field

from app.domain.enums.draft_status import DraftStatus


class DraftCreateRequest(BaseModel):

    email_id: UUID
    instructions: str | None = Field(default=None, max_length=2000)


class DraftUpdateRequest(BaseModel):
    body: str


class DraftResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    email_id: UUID
    body: str
    status: DraftStatus
    created_at: datetime
    updated_at: datetime